File: library/src/main/java/no/wtw/android/architectureutils/view/ListItemView.java
package no.wtw.android.architectureutils.view;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import org.androidannotations.annotations.EViewGroup;
import org.androidannotations.annotations.ViewById;

import no.wtw.android.architectureutils.adapter.ViewWrapper;
import no.wtw.android.architectureutils.model.Listable;

@EViewGroup(resName = "list_item_view")
public class ListItemView extends RelativeLayout implements ViewWrapper.Binder<Listable> {

    @ViewById(resName = "icon")
    protected ImageView iconView;
    @ViewById(resName = "title")
    protected TextView titleView;
    @ViewById(resName = "subtitle")
    protected TextView subTitleView;

    public ListItemView(Context context) {
        super(context);
    }

    public ListItemView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ListItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void bind(Listable data) {
        iconView.setImageResource(data.getIconResourceId(getContext()));
        iconView.setVisibility(data.getIconResourceId(getContext()) == 0 ? GONE : VISIBLE);
        titleView.setText(data.getTitle(getContext()));
        subTitleView.setText(data.getSubTitle(getContext()));
        subTitleView.setVisibility(TextUtils.isEmpty(data.getSubTitle(getContext())) ? GONE : VISIBLE);
    }

}

File: library/src/main/java/no/wtw/android/architectureutils/view/ListItemView.kt
package no.wtw.android.architectureutils.view

import android.content.Context
import android.text.TextUtils
import android.util.AttributeSet
import android.widget.RelativeLayout
import androidx.core.graphics.Insets
import no.wtw.android.architectureutils.adapter.ViewWrapper
import no.wtw.android.architectureutils.databinding.ListItemViewBinding
import no.wtw.android.architectureutils.model.Listable

open class ListItemView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0,
) : RelativeLayout(context, attrs, defStyleAttr),
    ViewWrapper.Binder<Listable> {

    protected val viewBinding by viewBinding<ListItemViewBinding>()

    override fun bind(data: Listable, insets: Insets) {
        viewBinding.apply {
            clickableRoot.setPadding(insets.left, insets.top, insets.right, insets.bottom)
            icon.setImageResource(data.getIconResourceId(context))
            icon.visibility = if (data.getIconResourceId(context) == 0) GONE else VISIBLE
            title.text = data.getTitle(context)
            subtitle.text = data.getSubTitle(context)
            subtitle.visibility = if (TextUtils.isEmpty(data.getSubTitle(context))) GONE else VISIBLE
        }
    }

}